internal/resolution/variables/bundle.go

```diff
@@ -1,42 +1,107 @@
 package variables
 
 import (
+	"encoding/json"
+	"fmt"
+	"sync"
+
 	"github.com/operator-framework/deppy/pkg/deppy"
 	"github.com/operator-framework/deppy/pkg/deppy/constraint"
 	"github.com/operator-framework/deppy/pkg/deppy/input"
+	"github.com/operator-framework/operator-registry/alpha/property"
+
+	bsemver "github.com/blang/semver/v4"
+)
 
-	olmentity "github.com/operator-framework/operator-controller/internal/resolution/entities"
+const PropertyBundlePath = "olm.bundle.path"
+const PropertyBundleChannelEntry = "olm.bundle.channelEntry"
+const PropertyBundleMediaType = "olm.bundle.mediatype"
+
+type MediaType string
+
+const (
+	MediaTypePlain    = "plain+v0"
+	MediaTypeRegistry = "registry+v1"
 )
 
+// ----
+
+type propertyRequirement bool
+
+const (
+	required propertyRequirement = true
+	optional propertyRequirement = false
+)
+
+type PackageRequired struct {
+	property.PackageRequired
+	SemverRange *bsemver.Range `json:"-"`
+}
+
+type GVK property.GVK
+
+type ChannelEntry struct {
+	Name     string `json:"name"`
+	Replaces string `json:"replaces"`
+	// Skips and skipRange will probably go here as well
+}
+
+func (g GVK) String() string {
+	return fmt.Sprintf(`group:"%s" version:"%s" kind:"%s"`, g.Group, g.Version, g.Kind)
+}
+
+type GVKRequired property.GVKRequired
+
+func (g GVKRequired) String() string {
+	return fmt.Sprintf(`group:"%s" version:"%s" kind:"%s"`, g.Group, g.Version, g.Kind)
+}
+
+func (g GVKRequired) AsGVK() GVK {
+	return GVK(g)
+}
+
 var _ deppy.Variable = &BundleVariable{}
 
 type BundleVariable struct {
 	*input.SimpleVariable
-	bundleEntity *olmentity.BundleEntity
-	dependencies []*olmentity.BundleEntity
-}
+	ID           deppy.Identifier
+	Properties   map[string]string
+	dependencies []*BundleVariable
 
-func (b *BundleVariable) BundleEntity() *olmentity.BundleEntity {
-	return b.bundleEntity
+	// these properties are lazy loaded as they are requested
+	bundlePackage    *property.Package
+	providedGVKs     []GVK
+	requiredGVKs     []GVKRequired
+	requiredPackages []PackageRequired
+	channel          *property.Channel
+	channelEntry     *ChannelEntry
+	semVersion       *bsemver.Version
+	bundlePath       string
+	mediaType        string
+	mu               sync.RWMutex
```

Comment:
Wondering if we should separate bundle variable (meant for deppy resolution) from bundle data struct holding data from catalog.